core/src/main/scala/dagr/core/cmdline/DagrCoreMain.scala

```diff
@@ -239,8 +199,94 @@ class DagrCoreMain(
+  def makeItSo(args: Array[String], packageList: List[String] = DagrCoreMain.getPackageList, includeHidden: Boolean = false): Int = {
+    // Initialize color options
+    TermCode.printColor = DagrCoreMain.optionallyConfigure[Boolean](Configuration.Keys.ColorStatus).getOrElse(true)
+
+    // Load any dagr scripts - assumes the option is "--scripts" only
+    loadScripts(args)
+
+    val startTime = System.currentTimeMillis()
+    val packages = Sopt.find[Pipeline](packageList, includeHidden=includeHidden)
+    val exit      = Sopt.parseCommandAndSubCommand[DagrCoreArgs,Pipeline](name, args, packages) match {
+      case Sopt.Failure(usage) =>
+        System.err.print(usage())
+        1
+      case Sopt.CommandSuccess(cmd) =>
+        unreachable("CommandSuccess should never be returned by parseCommandAndSubCommand.")
+      case Sopt.SubcommandSuccess(dagr, pipeline) =>
+        val name = pipeline.getClass.getSimpleName
+        try {
+          dagr.configure(pipeline, Some(args.mkString))
+          val name = Configuration.commandLineName(this.name)
+          printStartupLines(name, args)
+          val numFailed = dagr.execute(pipeline)
+          if (numFailed!= 0) logger.fatal("Execution failed!")
+          printEndingLines(startTime, name, true)
+          numFailed
+        }
+        catch {
+          case ex: Throwable =>
+            printEndingLines(startTime, name, false)
+            throw ex
+        }
+    }
+
+    exit
+  }
+
+  /** Prints a line of useful information when a tool starts executing. */
+  protected def printStartupLines(tool: String, args: Array[String]): Unit = {
+    val version    = CommandLineProgramParserStrings.version(getClass, color=false).replace("Version: ", "")
+    val host       = InetAddress.getLocalHost.getHostName
+    val user       = System.getProperty("user.name")
+    val jreVersion = System.getProperty("java.runtime.version")
+    logger.info(s"Executing $tool from $name version $version as $user@$host on JRE $jreVersion")
+  }
+
+  /** Prints a line of useful information when a tool stops executing. */
+  protected def printEndingLines(startTime: Long, name: String, success: Boolean): Unit = {
+    val elapsedMinutes: Double = (System.currentTimeMillis() - startTime) / (1000d * 60d)
+    val elapsedString: String = new DecimalFormat("#,##0.00").format(elapsedMinutes)
+    val verb = if (success) "completed" else "failed"
+    logger.info(s"$name $verb. Elapsed time: $elapsedString minutes.")
+  }
+
+  /** Loads the various dagr scripts and puts them on the classpath. */
+  private def loadScripts(args: Array[String]): Unit = {
+    val tokenizer = new ArgTokenizer(args, argFilePrefix=Some("@"))
+    val collator = new ArgTokenCollator(tokenizer)
+    collator.filter(_.isSuccess).foreach {
+      case Success(ArgOptionAndValues(name: String, values: Seq[String])) if name == "scripts" =>
+        val scripts = values.map(PathUtil.pathTo(_)).filter { path =>
```

Comment:
This is a little weird to me.  Why not just filter with `Files.isReadable(_)` instead of this big try/catch?  Also, I think it would be nicer to warn or even outright fail if any provided scripts dir is unreadable.